src/clients/chunker.rs

```diff
@@ -108,28 +125,32 @@ impl Client for ChunkerClient {
         "chunker"
     }
 
-    async fn health(&self) -> HealthCheckResult {
-        let mut client = self.health_client.clone();
-        let response = client
-            .check(HealthCheckRequest { service: "".into() })
-            .await;
+    async fn health(&self) -> Result<HealthCheckResult, Error> {
+        let request =
+            grpc_request_with_headers(HealthCheckRequest { service: "".into() }, HeaderMap::new());
+        let response: Result<HealthCheckResponse, Error> =
+            async { grpc_call!(self.health_client, request, HealthClient::check) }.await;
         let code = match response {
-            Ok(_) => Code::Ok,
-            Err(status) if matches!(status.code(), Code::InvalidArgument | Code::NotFound) => {
-                Code::Ok
-            }
-            Err(status) => status.code(),
+            Ok(_) => StatusCode::OK,
```

Comment:
grpc_call caused things to need to be rewritten but the logic should be the same, will see if I can reduce/clean up these changes